src/Tasks/CreateManifestResourceName.cs

```diff
@@ -146,8 +146,19 @@ CreateFileStream createFileStream
                     string fileName = resourceFile.ItemSpec;
                     string dependentUpon = resourceFile.GetMetadata(ItemMetadataNames.dependentUpon);
 
-                    // If opted into convention and no DependentUpon metadata, reference "<filename>.cs" if it exists.
-                    if (UseDependentUponConvention && string.IsNullOrEmpty(dependentUpon))
+                    string fileType = resourceFile.GetMetadata("Type");
+
+                    // If it has a file type metadata is "Resx"
+                    bool isResxFile = (!string.IsNullOrEmpty(fileType) && fileType == "Resx");
+
+                    //fall back onto the extension
```

Comment:
I'm not familiar with when Type=Resx is set, is that up to the user or to targets that precede this do it?

Reply:
Yes, please document the current state at least in the PR and maybe in the comment here.

Reply:
Users can manually set the `Type` metadata to `Resx` or `Non-Resx`. If that metadata isn't set, it is automatically populated by the time the task runs. Currently trying to track down when it gets set.

Reply:
It's set through the `SplitResourcesByCulture` target [here](https://github.com/microsoft/msbuild/blob/master/src/Tasks/Microsoft.Common.CurrentVersion.targets#L2940)

```xml
      <EmbeddedResource Include="@(_MixedResourceWithNoCulture);@(_MixedResourceWithCulture)" Condition="'%(Extension)'=='.resx' or '%(Extension)'=='.restext'">
        <Type Condition="'%(_MixedResourceWithNoCulture.Type)'=='' and '%(_MixedResourceWithCulture.Type)'==''">Resx</Type>
      </EmbeddedResource>
      <EmbeddedResource Include="@(_MixedResourceWithNoCulture);@(_MixedResourceWithCulture)" Condition="'%(Extension)'!='.resx' and '%(Extension)'!='.restext'">
        <Type Condition="'%(_MixedResourceWithNoCulture.Type)'=='' and '%(_MixedResourceWithCulture.Type)'==''">Non-Resx</Type>
      </EmbeddedResource>
```

Reply:
My follow up here is do we need this fallback to extension, then? It seems like we should expect Type to be set?

Reply:
Yeah, I think I now lean toward "assume it's set and if it's not, no convention" as well. I'll push that.